static-instrumenter/agent-instrumenter/build.gradle.kts

```diff
@@ -1,12 +1,138 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
 plugins {
   id("otel.java-conventions")
+  id("com.github.johnrengelman.shadow") version "7.1.2"
+  id("org.unbroken-dome.test-sets") version "4.0.0"
 }
 
 description = "OpenTelemetry Java Static Instrumentation Agent"
 
+val javaagentLibs: Configuration by configurations.creating {
+  isCanBeResolved = true
+  isCanBeConsumed = false
+}
+
+val bootstrapLibs: Configuration by configurations.creating
+configurations.getByName("implementation").extendsFrom(bootstrapLibs)
+
+val javaagent: Configuration by configurations.creating
+configurations.getByName("implementation").extendsFrom(javaagent)
+
 dependencies {
+  annotationProcessor("com.google.auto.service:auto-service")
+  compileOnly("com.google.auto.service:auto-service")
+
   implementation("org.slf4j:slf4j-api")
   runtimeOnly("org.slf4j:slf4j-simple")
+
+  // TODO: remove local jar once agent with hook is released
+  javaagent(files("libs/opentelemetry-javaagent.jar"))
+  // javaagent("io.opentelemetry.javaagent:opentelemetry-javaagent")
+
+  bootstrapLibs(project(":static-instrumenter:common"))
+  javaagentLibs(project(":static-instrumenter:agent-extension"))
+}
+
+testSets {
+  create("integrationTest")
+}
+
+tasks {
+
+  val relocateJavaagentLibs by registering(ShadowJar::class) {
+    configurations = listOf(javaagentLibs)
+    duplicatesStrategy = DuplicatesStrategy.FAIL
+    archiveFileName.set("javaagentLibs-relocated.jar")
+  }
+
+  val isolateJavaagentLibs by registering(Copy::class) {
+    dependsOn(relocateJavaagentLibs)
+    isolateClasses(relocateJavaagentLibs.get().outputs.files)
+    into("$buildDir/isolated/javaagentLibs")
+  }
+
+  shadowJar {
+    configurations = listOf(javaagent, bootstrapLibs)
+
+    dependsOn(isolateJavaagentLibs)
+    from(isolateJavaagentLibs.get().outputs)
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+    manifest {
+      attributes.put("Main-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.main.Main")
+      attributes.put("Agent-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.OpenTelemetryStaticAgent")
+      attributes.put("Premain-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.OpenTelemetryStaticAgent")
+      attributes.put("Can-Redefine-Classes", "true")
+      attributes.put("Can-Retransform-Classes", "true")
+      attributes.put("Implementation-Vendor", "OpenTelemetry")
+      attributes.put("Implementation-Version", "demo-${project.version}")
+    }
+  }
+
+  val mainShadowJar by registering(Jar::class) {
+    archiveFileName.set("opentelemetry-static-agent.jar")
+    from(zipTree(shadowJar.get().archiveFile))
+
+    manifest {
+      attributes(shadowJar.get().manifest.attributes)
+    }
+  }
+
+  // TODO: the resulting jar contains both inst and unpacked inst,
+  //  but we need only the unpacked inst here
+  val createNoInstAgent by registering(ShadowJar::class) {
```

Comment:
Done. On a similar note, I think we should change the name of `agent-instrumenter` to `static-agent`, since it no longer instruments the agent but provides a distro of the agent. WDYT?

Reply:
Yeah, it's a good call 👍